OpenSim/Common/Component.h

```diff
@@ -2686,26 +2696,26 @@ void ComponentListIterator<T>::advanceToNextValidComponent() {
 template<class C>
 void Connector<C>::findAndConnect(const Component& root) {
  
-    const std::string& path = getConnecteeName();
+    ComponentPath path(getConnecteeName());
     const C* comp = nullptr;
 
     try {
-        if (path[0] == '/') { //absolute path name
-            comp =  &root.template getComponent<C>(path);
+        if (path.isAbsolute()) {
+            comp =  &root.template getComponent<C>(path.getString());
```

Comment:
I am now tempted to make `getComponent<C>()` take a `ComponentPath` instead of a `string`. Next PR.